pulpcore/plugin/viewsets/content.py

```diff
@@ -43,20 +43,19 @@ def create(self, request):
         serializer.is_valid(raise_exception=True)
 
         task_payload = {k: v for k, v in request.data.items()}
-        file_content = task_payload.pop("file", None)
 
+        file_content = task_payload.pop("file", None)
         temp_file = PulpTemporaryFile.init_and_validate(file_content)
         temp_file.save()
 
-        resources = []
-        repository = serializer.validated_data.get("repository")
-        if repository:
-            resources.append(repository)
+        exclusive_resources = [
+            item for item in (serializer.validated_data.get(key) for key in ("repository")) if item
+        ]
```

Comment:
```suggestion
        exclusive_resources = [
            item for item in (serializer.validated_data.get(key) for key in ("repository",)) if item
        ]
```
You are iterating over the list of characters. Is this complex list comprehension necessary?

Reply:
Thanks! Not it's probably a bit much.

Reply:
I wanted to bring the two viewsets closer together in the way they are implemented.